inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -87,21 +88,45 @@ bool InLaneCruisingPlugin::plan_trajectory_cb(cav_srvs::PlanTrajectoryRequest& r
 
   ROS_DEBUG_STREAM("PlanTrajectory");
 
-  cav_msgs::TrajectoryPlan trajectory;
-  trajectory.header.frame_id = "map";
-  trajectory.header.stamp = ros::Time::now();
-  trajectory.trajectory_id = boost::uuids::to_string(boost::uuids::random_generator()());
+  cav_msgs::TrajectoryPlan initial_trajectory;
+  initial_trajectory.header.frame_id = "map";
+  initial_trajectory.header.stamp = ros::Time::now();
+  initial_trajectory.trajectory_id = boost::uuids::to_string(boost::uuids::random_generator()());
+
+  initial_trajectory.trajectory_points = compose_trajectory_from_centerline(downsampled_points, req.vehicle_state); // Compute the trajectory
+  initial_trajectory.initial_longitudinal_velocity = std::max(req.vehicle_state.longitudinal_vel, config_.minimum_speed);
 
-  trajectory.trajectory_points = compose_trajectory_from_centerline(downsampled_points, req.vehicle_state); // Compute the trajectory
-  trajectory.initial_longitudinal_velocity = std::max(req.vehicle_state.longitudinal_vel, config_.minimum_speed);
 
   if (config_.enable_object_avoidance)
-  {
-    resp.trajectory_plan = obj_.update_traj_for_object(trajectory, wm_, req.vehicle_state.longitudinal_vel);
+  { 
+    ROS_DEBUG_STREAM("Modifying Trajectory to Yield");
+    // compose yield service request
+    yield_srv_.request.vehicle_state = req.vehicle_state;
+    yield_srv_.request.initial_trajectory_plan = initial_trajectory;
+
+
+    if (yield_called_)
```

Comment:
Would you be able to explain what's going on with this service call pattern? It looks like the trajectory is being passed to yield in the `run()` function every tick, then retrieved here in the `plan_trajectory_cb()`? Is there an advantage to doing it that way over simply calling the Yield service at the end of `plan_trajectory_cb()` and then returning the resulting trajectory?